test/test_nn.py

```diff
@@ -6869,6 +6869,56 @@ def test_convert_sync_batchnorm(self):
                 self.assertEqual(layer.state_dict()[key].device, converted_layer.state_dict()[key].device)
                 self.assertEqual(layer.state_dict()[key], converted_layer.state_dict()[key])
 
+    @unittest.skipIf(not TEST_CUDA, "CUDA not available")
+    def test_sync_batchnorm_zero_count(self):
+        device = 'cuda'
+        count = torch.tensor([0, 5, 0], dtype=torch.int32, device=device)
+        n = count.sum()
+        eps = 1e-6
+        momentum = 1.
+
+        input = [torch.randn(c, 96, 112, 112, device=device) for c in count]
+        running_mean = torch.rand(96, device=device)
+        running_var = torch.rand(96, device=device)
+
+        mean_all = []
+        invstd_all = []
+        for i in input:
+            if i.numel() > 0:
+                m, iv = torch.batch_norm_stats(i, eps)
+                mean_all.append(m)
+                invstd_all.append(iv)
+            else:
+                mean_all.append(torch.zeros([96], device=device))
+                invstd_all.append(torch.ones([96], device=device))
+
+        mean_all = torch.stack(mean_all)
+        invstd_all = torch.stack(invstd_all)
+
+        mean_batch, inv_std_batch = torch.batch_norm_stats(torch.cat(input, dim=0), eps=eps)
+        running_mean_batch = (1 - momentum) * running_mean + momentum * mean_batch
+        running_var_batch = (1 - momentum) * running_var + momentum * n / (n - 1) / inv_std_batch ** 2
+
+        for i in input:
+            running_mean_ = running_mean.clone()
+            running_var_ = running_var.clone()
+
+            mean_, inv_std_ = torch.batch_norm_gather_stats_with_counts(
+                i,
+                mean_all,
+                invstd_all,
+                running_mean_,
+                running_var_,
+                momentum,
+                eps,
+                count.to(torch.get_default_dtype())
+            )
+
+            self.assertEqual(mean_, mean_batch)
+            self.assertEqual(inv_std_, inv_std_batch)
+            self.assertEqual(running_mean_, running_mean_batch)
+            self.assertEqual(running_var_, running_var_batch, atol=1.3e-6, rtol=1e-5)
```

Comment:
Assert fails with default precision target, it looks like a precision issue.
Alternately this could be the c expected value which has an error.
Or my modification messed something up, but I find it unlikely that running variance would be the sole victim.